### ISSUE TITLE
[tabular] Kaggle GPU leads to exception with `best_quality`

### ISSUE THREAD

**Issue by Innixma** [autogluon/autogluon#4164]
An exception is raised if using a GPU Kaggle notebook and specifying `best_quality` preset.

[User Report](https://www.kaggle.com/competitions/playground-series-s4e5/discussion/499495#2789980) using a Kaggle P100 GPU Notebook.

This is probably due to dynamic stacking's sub-fit process in Ray. It might also be due to any usage of Ray. We should test both on Kaggle and locally.

Error log:

```
Starting holdout-based sub-fit for dynamic stacking. Context path is: AutogluonModels/ag-20240501_091619/ds_sub_fit/sub_fit_ho.
/opt/conda/lib/python3.10/site-packages/requests/__init__.py:109: RequestsDependencyWarning: urllib3 (2.1.0) or chardet (None)/charset_normalizer (3.3.2) doesn't match a supported version!
  warnings.warn(
2024-05-01 09:16:19,705    INFO util.py:124 -- Outdated packages:
  ipywidgets==7.7.1 found, needs ipywidgets>=8
Run `pip install -U ipywidgets`, then restart the notebook server for rich notebook output.
2024-05-01 09:16:21,761    ERROR services.py:1330 -- Failed to start the dashboard , return code -11
2024-05-01 09:16:21,763    ERROR services.py:1355 -- Error should be written to 'dashboard.log' or 'dashboard.err'. We are printing the last 20 lines for you. See 'https://docs.ray.io/en/master/ray-observability/ray-logging.html#logging-directory-structure' to find where the log file is.
2024-05-01 09:16:21,765    ERROR services.py:1399 -- 
The last 20 lines of /tmp/ray/session_2024-05-01_09-16-19_717749_24/logs/dashboard.log (it contains the error message from the dashboard): 
2024-05-01 09:16:21,701    INFO head.py:254 -- Starting dashboard metrics server on port 44227
```

**Comment by mglowacki100** [comment]
It looks similar to this issue: https://github.com/autogluon/autogluon/issues/3365

**Comment by Innixma** [comment]
Fixed in #4210
